tests/PHPStan/Analyser/data/bug-11912.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Bug11912;
+
+/**
+ * @param array<string, mixed> $results
+ * @param list<string> $names
+ */
+function appendResults(array $results, array $names): bool {
+	// Make sure 'names' comes first in array
+	$results = ['names' => $names] + $results;
+	\PHPStan\Testing\assertType("list<string>", $results['names']);
```

Comment:
This does not need to be in AnalyserIntegrationTest. Simply move it to `tests/PHPStan/Analyser/nsrt/bug-11912.php` and it will be automatically run with NodeScopeResolverTest.